packages/live-share-inking/src/core/Internals.ts

```diff
@@ -0,0 +1,365 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the Microsoft Live Share SDK License.
+ */
+import { v4 as uuid } from "uuid";
+import { IPoint, IPointerPoint, IRect, ISegment } from ".";
+
+/**
+ * Pre-calculated Pi x 2.
+ */
+export const TWO_PI: number = Math.PI * 2;
+
+const EPSILON = 0.000001;
+
+/**
+* Generates a unique Id.
+* @returns A unique Id.
+*/
+export function generateUniqueId(): string {
+    return uuid();
+}
+
+/**
+ * Converts a PointerEvent into an IPointerPoint.
+ * @param e The pointer event to convert.
+ * @returns An IPointerPoint object.
+ */
+export function pointerEventToPoint(e: PointerEvent): IPointerPoint {
+    return {
+        x: e.offsetX,
+        y: e.offsetY,
+        pressure: e.pressure
+    };
+}
+
+/**
+ * Determines if a number is within a randge.
+ * @param n The number to check.
+ * @param r1 The first range boundary.
+ * @param r2 The second range boundary.
+ * @returns `true` if `n` is between `r1` and `r2`, `false` otherwise.
+ */
+export function isInRange(n: number, r1: number, r2: number): boolean {
+    const adjustedMin = Math.min(r1, r2) - EPSILON;
+    const adjustedMax = Math.max(r1, r2) + EPSILON;
+
+    return n >= adjustedMin && n <= adjustedMax;
+}
+
+/**
+ * @hidden
+ * Defines a quad, i.e. a polygon with 4 sides.
+ */
+export interface IQuad {
+    p1: IPoint;
+    p2: IPoint;
+    p3: IPoint;
+    p4: IPoint;
+}
+
+/**
+ * @hidden
+ * Defines a segment in a path that joins circles
+ * or rectangles.
+ */
+export interface IQuadPathItem {
```

Comment:
What do you think of the name IQuadPathSegment?

Reply:
Sounds better, will change!